crates/compiler/can/src/annotation.rs

```diff
@@ -17,20 +17,51 @@ use roc_types::types::{
 pub struct Annotation {
     pub typ: Type,
     pub introduced_variables: IntroducedVariables,
-    pub references: VecSet<Symbol>,
+    pub references: AnnotationReferences,
     pub aliases: VecMap<Symbol, Alias>,
 }
 
+#[derive(Clone, Debug)]
+pub struct AnnotationReferences {
+    symbols: VecSet<Symbol>,
+    qualified: Vec<QualifiedReference>,
+}
+
+impl AnnotationReferences {
+    pub fn new() -> Self {
+        Self {
+            symbols: Default::default(),
+            qualified: Default::default(),
+        }
+    }
+
+    pub fn insert(&mut self, symbol: Symbol, qualified: QualifiedReference) {
+        if !self.symbols.insert(symbol) {
+            self.qualified.push(qualified);
+        }
+    }
+
+    pub fn insert_lookups(&self, references: &mut References) {
+        for (symbol, qualified) in self.symbols.iter().zip(&self.qualified) {
+            references.insert_type_lookup(*symbol, *qualified);
+        }
+    }
```

Comment:
It feels a little strange to have both this API exposed publicly, and `Annotation#add_to`. I am a little bit worried about drift where this is called, and the other isn't. I wonder if there's a way to only use the `Annotation#add_to` API? Or better yet, have some way to consume an `Annotation` into a context instead of having to call an API on the Annotation itself.

Not a blocking comment, just something for thought.

Reply:
Right. I followed the existing pattern here. The reason `insert_lookups` is needed is that not all the functions take `Output` and therefore cannot call `Annotation#add_to`. That said, if move `References` to `Env` as you suggested, we wouldn't have this problem and we wouldn't have to union `Output`.

Reply:
Makes sense. This can be resolved in the future.